### PR TITLE
Fix issues with "update experts on existing trees" API endpoint

### PR DESCRIPTION
**Pull request by zjmiller**

### CODE REVIEW

**Comment by zjmiller** [comment]
Made sure transaction always got properly passed, and added a few missing `await`s.

File: server/lib/schema/index.ts [hunk 1]
```diff
@@ -2189,7 +2189,7 @@ const schema = new GraphQLSchema({
                     );
                   }
 
-                  tree.$add("oracle", user, transaction);
+                  await tree.$add("oracle", user, { transaction });
```
Comment (andrewschreiber) [inline server/lib/schema/index.ts]: I wonder if there was a race condition here previously that was causing the inconsistent behavior.

File: server/lib/schema/index.ts [hunk 2]
```diff
@@ -2204,14 +2204,16 @@ const schema = new GraphQLSchema({
...
 
                   const oracleRelation: UserTreeOracleRelation | null = await UserTreeOracleRelation.findOne(
                     {
                       where: {
                         UserId: user.id,
                         TreeId: tree.id,
                       },
+
+                      transaction,
```
Comment (andrewschreiber) [inline server/lib/schema/index.ts]: Why pass the transaction here?

File: server/lib/schema/index.ts (hunk 1 above)
Comment (andrewschreiber) [reply server/lib/schema/index.ts]: Good catch on wrapping the transaction with an object.

File: server/lib/schema/index.ts (hunk 2 above)
Comment (andrewschreiber) [inline server/lib/schema/index.ts]: It looks like the field expects a `FindOptions` object, of which `transaction` is one property.  (Specifically, `FindOptions` is an interface that extends `QueryOptions`, which extends `Transactionable`, which has a `transaction` property). So I'd expect this to need to be surrounded by curly braces as well.

Comment (andrewschreiber) [reply server/lib/schema/index.ts]: Somewhat disappointed TypeScript didn't catch the `transaction type errors I made in the past PR. I wonder if there is a stricter check we could use. It shouldn't have accepted a `Transaction` object when it was looking for a `FindOptions` object...

**Pull request merged**